From 48e49debd8a786fe715bc2ecca21797a9f3be246 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Thu, 11 Feb 2016 14:22:05 -0800
Subject: [PATCH 1/6] camera: lm3642: check for possible NULL dereference

Be sure that lm3642 data struct isn't NULL before dereferencing.

Change-Id: I0b8b1fc86e31d8e6dfd07c99f3112d4f3f565f71
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19701
Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
Reviewed-on: https://android.intel.com:443/469507
---
 drivers/external_drivers/camera/drivers/media/i2c/lm3642.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/lm3642.c b/drivers/external_drivers/camera/drivers/media/i2c/lm3642.c
index 98c42dd..b94808e 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/lm3642.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/lm3642.c
@@ -780,10 +780,10 @@ void *lm3642_platform_data_func(struct i2c_client *client)
 
 	if (ACPI_COMPANION(&client->dev)) {
 		lm3642_dev = gmin_camera_platform_data(sd, 0, 0);
-		platform_data.power_ctrl = lm3642_dev->v1p8_ctrl;
-	}
 
-	if (ACPI_COMPANION(&client->dev)) {
+		if (lm3642_dev)
+			platform_data.power_ctrl = lm3642_dev->v1p8_ctrl;
+
 		platform_data.gpio_strobe =
 			desc_to_gpio(gpiod_get_index(&(client->dev),
 							"lm3642_gpio0", 0));
-- 
1.9.1

